internal/bot/bot.go

```diff
@@ -24,30 +27,64 @@ type Bot struct {
 }
 
 // Answer takes a user input and executes a transition on the FSM if possible
-func (b *Bot) Answer(question *query.Question) ([]query.Answer, error) {
-	if !b.Store.Exists(question.Sender) {
-		b.Store.Set(question.Sender, fsm.NewFSM())
+func (b *Bot) Answer(receiveMsg *messages.Receive) ([]query.Answer, error) {
+	isExistingConversation := b.Store.Exists(receiveMsg.Conversation())
+
+	if !isExistingConversation {
+		b.Store.Set(receiveMsg.Conversation(), fsm.NewFSM())
 	}
 
-	cmd, _ := b.Classifier.Predict(question.Text)
+	cmd, _ := b.Classifier.Predict(receiveMsg.Question.Text)
 
-	machine := b.Store.Get(question.Sender)
+	machine := b.Store.Get(receiveMsg.Conversation())
 
 	previousState := machine.State
 
-	reply, ext := machine.ExecuteCmd(cmd, question.Text, b.Domain)
+	answers, ext, err := machine.ExecuteCmd(cmd, receiveMsg.Question.Text, b.Domain)
+	if err != nil {
+		switch e := err.(type) {
+		case *fsm.ErrUnsureCommand:
+			if b.Config.ShouldReplyUnsure(isExistingConversation) {
```

Comment:
Here we use the `ReplyUnsure` configuration to see if we should return an `Unsure` message.